### DISCUSSION TITLE
Insert Join

### DISCUSSION THREAD

**Discussion by erwanvivien** [SeaQL/sea-orm#1646] [Q&A]
## Motivation

I have two tables:
```rust
User { id: Uuid, validated: bool, .. }
Books { user_id: Uuid, name: String, .. } 
```

And I would like to assert that user is validated while creating new books.
It allows me to have only one (possibly failing) query instead of two.

https://stackoverflow.com/a/15710598

## Proposed Solutions

I don't know

## Current Workarounds

Double query. Fetch then insert

**Comment by erwanvivien** [comment]
My correct SQL query would be:

```sql
INSERT INTO
    "Books" (
        "user_id",
        "name"
    )
SELECT
    'MY_ID',
    'Harry Potter'
WHERE EXISTS (
        SELECT 1
        FROM "User"
        WHERE
            "User"."id" = 'MY_ID'
            AND "User"."validated" = TRUE
    );
```

Which returns 
```
AffectedRows : 0
```

When user is not validated

**Comment by tyt2y3** [comment]
It should be possible to build such a query with SeaQuery, or with a raw SQL statement.

https://docs.rs/sea-query/latest/sea_query/query/struct.InsertStatement.html#method.select_from
